tinygrad/renderer/assembly.py

```diff
@@ -63,7 +63,7 @@ def uops_to_asm(lang:AssemblyLanguage, function_name:str, uops:UOpGraph) -> str:
         new = uops.add(UOps.ALU, dtypes.bool, (u,), arg=UnaryOps.NEG, insert_before=uops.uops.index(u)+1)
         replace[u] = new
       if u.arg == BinaryOps.CMPLT:
-        new = uops.add(UOps.ALU, dtypes.bool, (u.vin[0],), arg=UnaryOps.NEG, insert_before=uops.uops.index(u))
+        new = uops.add(UOps.ALU, dtypes.bool, (u.vin[0],), arg=UnaryOps.NEG, insert_before=uops.uops.index(u), simplify=False)
```

Comment:
Is this a fix, or masking an issue?

Reply:
If I understand this rewrite, it seems PTX understands a boolean algebra where `-x * y` is semantically `!x && y` for booleans. Python does not do this, so exec_alu would instead do `-int(x) * int(y)` which is wrong.

I don't think this is masking an issue because having exec_alu mimic this for non-PTX backends would be wrong.

Reply:
I don't understand. Which answer is correct? `simplify=False` is a hack

Reply:
I don't think wrong or correct describes either of them. Python decided that `False < True` was a valid expression and should resolve to `True`. This PTX IR has decided that `<` is not defined for booleans, so the workaround is to replace `x < y` with `-x * y`.

Python says that `-False * True` is `0` which is not the same as `False < True`, but PTX says thats that `-False * True` is the same as `!False && True` which is the same as Python's `False < True`.

`simplify=False` is not a hack; it is a way of dealing the diverging semantics of Python and PTX.

Reply:
Oh, PTX is wrong it sounds like.

False < True should evaluate to True